gcloud-java-pubsub/src/main/java/com/google/gcloud/pubsub/spi/PublisherApi.java

```diff
@@ -276,22 +94,38 @@ public Object extractNextToken(ListTopicSubscriptionsResponse payload) {
   // ========
 
   private final ManagedChannel channel;
-  private final ServiceApiSettings<MethodIdentifier> settings;
-  private final ImmutableMap<MethodIdentifier, ImmutableSet<Status.Code>> retryCodesConfig;
-  private final ImmutableMap<MethodIdentifier, RetryParams> retryParamsConfig;
+  private final List<AutoCloseable> closeables = new ArrayList<>();
+
+  private final ApiCallable<Topic, Topic> createTopicCallable;
+  private final ApiCallable<PublishRequest, PublishResponse> publishCallable;
+  private final ApiCallable<GetTopicRequest, Topic> getTopicCallable;
+  private final ApiCallable<ListTopicsRequest, ListTopicsResponse> listTopicsCallable;
+  private final ApiCallable<ListTopicsRequest, Iterable<Topic>> listTopicsIterableCallable;
+  private final ApiCallable<ListTopicSubscriptionsRequest, ListTopicSubscriptionsResponse>
+      listTopicSubscriptionsCallable;
+  private final ApiCallable<ListTopicSubscriptionsRequest, Iterable<String>>
+      listTopicSubscriptionsIterableCallable;
+  private final ApiCallable<DeleteTopicRequest, Empty> deleteTopicCallable;
 
   // ===============
   // Factory Methods
   // ===============
 
   /**
+   * Constructs an instance of PublisherSettings with default settings.
+   */
+  public static PublisherSettings newSettings() {
```

Comment:
It's not strictly necessary. I put it there to make all functionality discoverable through PublisherApi.